src/Lykke.AzureStorage/Tables/Templates/Index/AzureIndex.cs

```diff
@@ -123,28 +122,38 @@ public static AzureMultiIndex Create(string partitionKey, string rowKey, params
 
 			return result;
 		}
-
-
 	}
 
 	public static class AzureIndexUtils
 	{
+	    public static Tuple<string, string> ToTuple(this IAzureIndex src)
+	    {
+	        return Tuple.Create(src.PrimaryPartitionKey, src.PrimaryRowKey);
+	    }
+
 		public static Task<T> DeleteAsync<T>(this INoSQLTableStorage<T> tableStorage, IAzureIndex index) where T : ITableEntity, new()
 		{
 			return tableStorage.DeleteAsync(index.PrimaryPartitionKey, index.PrimaryRowKey);
 		}
 
+        /// <summary>
+        /// Gets data for array of indecies
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="tableStorage">Table storage</param>
+        /// <param name="indices">Array of indecies</param>
+        /// <param name="pieces"></param>
+        /// <param name="filter"></param>
+        /// <returns></returns>
 		public static async Task<IEnumerable<T>> GetDataAsync<T>(this INoSQLTableStorage<T> tableStorage,
 				 IEnumerable<IAzureIndex> indices, int pieces = 15, Func<T, bool> filter = null) where T : ITableEntity, new()
 		{
-			var idx = indices.ToArray();
-			if (idx.Length == 0)
-				return new T[0];
+            var idx = indices.ToArray();
```

Comment:
Use Any()  to check. ToArray() creates a copy of all values only for one element